### ISSUE TITLE
TAG column

### ISSUE THREAD

**Issue by Luca-noob** [influxdata/telegraf#15717]
### Please direct all support questsions to slack or the forums. Thank you.

Hi, I'm trying to export data from InfluxDB to Postgres.

I expected to have the '_field' column and the '_value' column also on pg, but I notice that a column is created for each tag present in the database, how can I solve this problem?

```
[[outputs.postgresql]]
  timestamp_column_name = "dthhrl"
  timestamp_column_type = "timestamp without time zone"
  create_templates = [
    ''' CREATE TABLE {{ .table }} ({{ .columns }});
	'''
  ]
```

**Comment by powersj** [comment]
> I expected to have the '_field' column and the '_value' column also on pg,

How are you querying InfluxDB with Telegraf? In general, I would expect any way we query to combine these into a key/value pair where the key is the column name and the value stored in that column.

**Comment by Luca-noob** [comment]
I use telegraf to take modbus data, write them to influxdb and then copy them to postgres. Actually on InfluxDB there is the value and field column, but the transposition on postgres creates the tag1 column with its value, tag2 with its value and so on

**Comment by powersj** [comment]
> Actually on InfluxDB there is the value and field column

Right, the whole _field and _value is due querying with Flux and the way it represents the results.

It sounds like you are after the [unpivot processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/unpivot), which would take something like:

> metric temp=42i,humidity=22i

and turn it into:

> metric,name=temp value=42i
> metric,name=humidity value=22i

If that is not what you are after please provide an example.

**Comment by Luca-noob** [comment]
In the meantime, thanks for the support Powerjs =)

I'm forwarding two screenshots to show you what I mean

![image](https://github.com/user-attachments/assets/4f75fee3-5bbb-48db-adbb-27b8a2046594)

![image](https://github.com/user-attachments/assets/39203d70-d58a-4ee8-a4f6-e261383d97b3)

**Comment by Luca-noob** [comment]
Ok the correct plugin is unpivot

```
[[outputs.postgresql]]
  connection = "XXXXXXX"
  timestamp_column_name = "dthhrl"
  timestamp_column_type = "timestamp with time zone"
  #fieldexclude = ["type"] don't work

[[processors.rename]]
	[[processors.rename.replace]] 
    measurement = "modbus"
    dest = "biggateway"
    [[processors.rename.replace]]
      tag = "name"
      dest = "ds_key"
    [[processors.rename.replace]]
      tag = "slave_id"
      dest = "ds_disp"	  

[[processors.unpivot]]
   tag_key = "ds_var"
   value_key = "value"
```
![image](https://github.com/user-attachments/assets/120afecb-27ff-413d-a824-73b11a44856f)

I ask you a further question.

How can I display the correct date without timezone, if I set it without timezone it writes the UTC+0 time to postgres.

I would like the value UTC+2 to be entered

**Comment by powersj** [comment]
> How can I display the correct date without timezone, if I set it without timezone it writes the UTC+0 time to postgres.

I'm not sure you can. Postgresql stores all values with a timestamp as UTC and without a timestamp it is assumed to be the local time zone and converted to UTC again.

**Comment by Luca-noob** [comment]
I noticed that unpivot inserts the data as int64 even if float32 and int16 values ​​are present, this makes the collected data unusable

**Comment by Luca-noob** [comment]
ok, little by little I'm getting around the problem (I'm learning a lot and I'm enjoying it), but I added some aggregations and this ruins all the work I've done

`[[processors.rename]]
	[[processors.rename.replace]] 
    measurement = "modbus"
    dest = "biggateway"
    [[processors.rename.replace]]
      tag = "name"
      dest = "ds_key"
    [[processors.rename.replace]]
      tag = "slave_id"
      dest = "ds_disp"	

##########################################################################	
 [[inputs.modbus]]
  name = "10.21.4.15:10502"
  timeout = "1s"
  controller = "tcp://10.21.4.15:10502"
  configuration_type = "request"
#-------------------------------------------------------------------------
  [[inputs.modbus.request]]
  slave_id = 10
  byte_order = "ABCD"
  register = "holding"
  fields = [
      { address=1001, name="voltage",      type="INT16"},
	  { address=1003, name="power",      type="FLOAT32"},
	  { address=1005, name = "Energy",     type = "INT64"}
]
  [[inputs.modbus.request]]
  slave_id = 15
  byte_order = "ABCD"
  register = "holding"
  fields = [
      { address=1001, name="voltage",      type="INT16"},
	  { address=1003, name="power",      type="FLOAT32"},
	  { address=1005, name = "Energy",     type = "INT64"}
]
###########################################################################
[[inputs.modbus]]
  name = "10.21.4.15:10503"
  timeout = "1s"
  controller = "tcp://10.21.4.15:10503"
  configuration_type = "request"
#-------------------------------------------------------------------------
  [[inputs.modbus.request]]
  slave_id = 15
  byte_order = "ABCD"
  register = "holding"
  fields = [
      { address=1001, name="voltage",      type="INT16"},
	  { address=1003, name="power",      type="FLOAT32"},
	  { address=1005, name = "Energy",     type = "INT64"}
]

[[processors.unpivot]]
   tag_key = "ds_var"
   value_key = "valor"
   
[[aggregators.basicstats]]
  period = "1m"
  drop_original = true
  tagpass={ds_var=["voltage", "power"]}
  stats = ["mean"]`

the result is this: I lose the correct tag name

![image](https://github.com/user-attachments/assets/e83f8075-6014-4bea-8b50-76a0d29613b4)

I apologize if I insist on using postgres, but a third application will interface with the same db

**Comment by Luca-noob** [comment]
I expect 'voltage' and 'power' or 'voltage_mean' and 'power_mean'

**Comment by powersj** [comment]
By default, processors are run after aggregators, that might be the cause here given you have two processor. You can try the `skip_processors_after_aggregators = true` agent config option.

**Comment by powersj** [comment]
I would greatly prefer you continue to ask questions in the slack or forums and not a closed issue as well. If you still have questions, please use one of them.
